provider/datagen/src/bin/datagen.rs

```diff
@@ -125,6 +125,66 @@ fn main() -> eyre::Result<()> {
                 .help("Whether to optimize Unicode property data structures for size (\"small\") or speed (\"fast\")")
                 .default_value("small"),
         )
+        .arg(
+            Arg::with_name("COLL_ROOT")
+                .long("coll-root")
+                .value_name("PATH")
+                .help(
+                    "Path to the icuexportdata coll directory. Download a \
+                    icuexportdata_uprops_*.zip file and point to the \
+                    'coll' subdirectory.\n\
+                    https://github.com/unicode-org/icu/releases",
+                )
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("COLL_DIA_ROOT")
+                .long("coll-dia-root")
+                .value_name("PATH")
+                .help(
+                    "Path to the icuexportdata dia directory. Download a \
+                    icuexportdata_uprops_*.zip file and point to the \
+                    'dia' subdirectory.\n\
+                    https://github.com/unicode-org/icu/releases",
+                )
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("COLL_JAMO_ROOT")
+                .long("coll-jamo-root")
+                .value_name("PATH")
+                .help(
+                    "Path to the icuexportdata dia directory. Download a \
+                    icuexportdata_uprops_*.zip file and point to the \
+                    'dia' subdirectory.\n\
+                    https://github.com/unicode-org/icu/releases",
+                )
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("COLL_META_ROOT")
+                .long("coll-meta-root")
+                .value_name("PATH")
+                .help(
+                    "Path to the icuexportdata collmeta directory. Download a \
+                    icuexportdata_uprops_*.zip file and point to the \
+                    'dia' subdirectory.\n\
+                    https://github.com/unicode-org/icu/releases",
+                )
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("COLL_REORD_ROOT")
+                .long("coll-reord-root")
+                .value_name("PATH")
+                .help(
+                    "Path to the icuexportdata collreord directory. Download a \
+                    icuexportdata_uprops_*.zip file and point to the \
+                    'dia' subdirectory.\n\
+                    https://github.com/unicode-org/icu/releases",
+                )
+                .takes_value(true),
+        )
```

Comment:
Is it really necessary to be able to specify all these directories separately? 
Adding more flags means there is more work for users to do, and it sounds like these directories are all in the uprops zip file, which already has a flag and download functionality.

Reply:
All the collator TOML files could go in a single directory if the collator-related provider copypaste learned to filter by the type of file. So far, on the provider side, I've just tried to do whatever lets me work on the collator algorithms.

Reply:
If you want to unblock collator work, please remove these arguments and hardcode the testdata path. We can deal with this API later.

Reply:
I have flattened these to a single directory for collation data.